lib/membrane_http_adaptive_stream/sink_bin.ex

```diff
@@ -172,11 +172,15 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
   end
 
   @impl true
-  def handle_pad_removed(Pad.ref(:input, ref), _ctx, state) do
+  def handle_pad_removed(Pad.ref(:input, ref), ctx, state) do
     children =
-      [
-        {:payloader, ref}
-      ] ++ if(state.mode != :muxed_av, do: [{:cmaf_muxer, ref}], else: [])
+      ([
+         {:payloader, ref}
+       ] ++ if(state.mode != :muxed_av, do: [{:cmaf_muxer, ref}], else: []))
+      |> Enum.filter(fn child_name ->
+        child_entry = Map.get(ctx.children, child_name)
+        child_entry != nil and !child_entry.terminating?
```

Comment:
`child_entry.terminating?` seems to be a boolean, so you probably should use `not` instead of `!`.

Also, you seem to have a lot of negation going on so I would consider `Enum.reject/2`
```suggestion
      |> Enum.reject(fn child_name ->
        child_entry = Map.get(ctx.children, child_name)
        is_nil(child_entry) or child_entry.terminating?
```